library/core/src/mem/mod.rs

```diff
@@ -1118,6 +1118,11 @@ impl<T> fmt::Debug for Discriminant<T> {
 ///
 /// [Reference]: ../../reference/items/enumerations.html#custom-discriminant-values-for-fieldless-enumerations
 ///
+/// The value of a [`Discriminant<T>`] is independent of any *lifetimes* in `T`. As such, reading
```

Comment:
```suggestion
/// The value of a [`Discriminant<T>`] is independent of any **free** *lifetimes* in `T`. As such, reading
```
It can be unsound to transmute `Discrimant<MyType<for<'a, 'b> fn(Inv<'a>, Inv<'b>)>>` to `Discrimant<MyType<for<'a> fn(Inv<'a>, Inv<'a>)>>`.

Reply:
(as a general procedural thought, I feel like this decision should have been at least co-owned by t-types)

Reply:
opened #118006

Reply:
My apologies about not looping in t-types. That team didn't exist when I first opened this PR. However, they should have been added while this PR was still under review.